python/src/internal.rs

```diff
@@ -30,12 +30,17 @@ use hudi::file_group::FileSlice;
 use hudi::table::builder::TableBuilder;
 use hudi::table::Table;
 
-macro_rules! vec_string_to_slice {
+fn convert_vec_to_slice(vec: &[(String, String, String)]) -> Vec<(&str, &str, &str)> {
+    vec.iter()
+        .map(|(a, b, c)| (a.as_str(), b.as_str(), c.as_str()))
+        .collect()
+}
+
+macro_rules! vec_to_slice {
     ($vec:expr) => {
-        &$vec.iter().map(AsRef::as_ref).collect::<Vec<_>>()
+        &convert_vec_to_slice(&$vec)[..]
     };
 }
```

Comment:
while you're here, can you move this util to a common module in hudi-core, with a basic UT covering it?

Reply:
I see that some modules in hudi core have utils.rs, I followed the same for this python crate for now

Reply:
@kazdy this is a generic util that should be placed under common for all modules to use. python does not mimic rust for every util, it imports when needed.

Reply:
ok, done